display/ssd1306.h

```diff
@@ -445,69 +461,122 @@ class StandardDisplayController : public DisplayControllerBase<Width, col_t>, Sa
     }
   }
 
+  void SB_On2() override {
+    if (IMG_out) {
+      ShowFileWithSoundLength(&IMG_out, font_config.ProffieOSOutImageDuration);
+    }
+  }
+
   void usb_connected() override {
     if (EscapeIdleIfNeeded() && current_effect_ == &IMG_idle) {
       // We are idle-looping, and usb is connected. Time to stop.
-      SetMessage("usb");
+      SetMessage("    usb\nconnected");
       SetScreenNow(SCREEN_MESSAGE);
     }
   }
 
-  void SB_Effect(EffectType effect, float location) override {
+  void ShowFileWithSoundLength(Effect* e, float duration) {
+    float dur = (duration == 0) ? SaberBase::sound_length * 1000 : duration;
+    ShowFile(e, round(dur));
+  }
+
+ void SB_Effect(EffectType effect, float location) override {
     switch (effect) {
       case EFFECT_BOOT:
-	if (IMG_boot) {
-	  ShowFile(&IMG_boot, font_config.ProffieOSFontImageDuration);
-	} else if (IMG_idle) {
-	  ShowFile(&IMG_idle, 3600000.0);
-	}
-	return;
+        if (IMG_boot) {
+          ShowFileWithSoundLength(&IMG_boot, font_config.ProffieOSBootImageDuration);
```

Comment:
This should SB_Effect2 to be guaranteed to work.

Reply:
moved.